test/remote/gateways/remote_deepstack_test.rb

```diff
@@ -0,0 +1,250 @@
+require 'test_helper'
+
+class RemoteDeepstackTest < Test::Unit::TestCase
+  def setup
+    @gateway = DeepstackGateway.new(fixtures(:deepstack))
+
+    # @gateway = DeepstackGateway.new(publishable_api_key: 'pk_test_XQS71KYAW9HW7XQOGAJIY4ENHZYZEO0C', app_id: '6652820b-6a7a-4d36-bc32-786e49da1cbd', shared_secret: 'ME1uVox0hrk7i87e7kbvnID38aC2U3X8umPH0D+BsVA=', sandbox: true)
+    @credit_card = credit_card
+    @amount = 100
+
+    @credit_card = ActiveMerchant::Billing::CreditCard.new(
+      number: '4111111111111111',
+      verification_value: '999',
+      month: '01',
+      year: '2029',
+      first_name: 'Bob',
+      last_name: 'Bobby'
+    )
+
+    @invalid_card = ActiveMerchant::Billing::CreditCard.new(
+      number: '5146315000000051',
+      verification_value: '999',
+      month: '01',
+      year: '2029',
+      first_name: 'Failure',
+      last_name: 'Fail'
+    )
+
+    address = {
+      address1: '123 Some st',
+      address2: '',
+      first_name: 'Bob',
+      last_name: 'Bobberson',
+      city: 'Some City',
+      state: 'CA',
+      zip: '12345',
+      country: 'USA',
+      phone: '1231231234',
+      email: 'test@test.com'
+    }
+
+    shipping_address = {
+      address1: '321 Some st',
+      address2: '#9',
+      first_name: 'Jane',
+      last_name: 'Doe',
+      city: 'Other City',
+      state: 'CA',
+      zip: '12345',
+      country: 'USA',
+      phone: '1231231234',
+      email: 'test@test.com'
+    }
+
+    @options = {
+      order_id: '1',
+      billing_address: address,
+      shipping_address: shipping_address,
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_successful_token
+    response = @gateway.gettoken(@credit_card, @options)
+    assert_success response
+
+    sale = @gateway.purchase(@amount, response.authorization, @options)
+    assert_success sale
+    assert_equal 'Approved', sale.message
+  end
+
+  def test_failed_token
+    response = @gateway.gettoken(@invalid_card, @options)
+    # puts response.message
+    assert_failure response
+    assert_equal 'InvalidRequestException: Card number is invalid.', response.message
+  end
+
+  # def test_successful_vault
+
+  #   response = @gateway.gettoken(@credit_card, @options)
+  #   assert_success response
+
+  #   vault = @gateway.store(response.authorization, @options)
+  #   assert_success vault
+
+  #   sale = @gateway.purchase(@amount, vault.authorization, @options)
+  #   assert_success sale
+
+  # end
```

Comment:
If the test does not pass but is something you want to keep could you provide a comment on why it is failing (ex: this feature is gated)

Reply:
Commented that feature is gated now :)